pkg/alertmanager/multitenant.go

```diff
@@ -848,6 +850,9 @@ func (am *MultitenantAlertmanager) setConfig(cfg amConfig) error {
 	// 2) then, submitted a non-working configuration (and we kept running the prev working config)
 	// 3) finally, the cortex AM instance is restarted and the running version is no longer present
 	if userAmConfig == nil {
+		if err != nil {
```

Comment:
I don't believe it's possible for both `userAmConfig` and `err` to be `nil` at the same time. In other words, an error must have occurred for `userAmConfig` to be `nil`.